### PR TITLE
Refresh of PyTorch Image Classification Multi-Node Distributed Data Parallel Training on CPU using Vertex Training with Custom Container Notebook

### PR DESCRIPTION
**Pull request by manuelamunategui**
This PR contains a notebook that demonstrates how to create a distributed PyTorch training job using Vertex SDK and custom containers. This can help your training job scale to handle a large amount of data.

- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/docs/contributing.md#code-quality-checks).
- [x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/docs/CODEOWNERS) file under `# Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

**Comment by manuelamunategui** [comment]
Notebook runs on Workbench, not Colab as it doesn't support custom containers yet

File: notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb [hunk 1]
```diff
@@ -0,0 +1,779 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>missing run in Colab</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>Vertex SDK -&gt; Vertex AI SDK for Python</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>It is a large database of handwritten digits that is commonly used for training various image processing systems.  -&gt; replace this with official description.</p><p> </p><p>http://yann.lecun.com/exdb/mnist/</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>how to use create -&gt; how to create</p><p> </p><p>Vertex SDK n-&gt; Vertex AI SDK</p><p> </p><p>Vertex Tensorboard -&gt; Vertex AI TensorBoard</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>Vertex SDK -&gt; Vertex AI SDK</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>REGION = "[your-region]"&nbsp;# @param {type: "string"}</p><p> </p><p>if REGION == "[your-region]":</p><p>&nbsp;&nbsp;REGION = "us-central1"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>make the first four cells a single cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <h2>Vertex Training using Vertex SDK and Custom Container -&gt; Vertex AI Training using a custom container</h2><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>should use the google artifact registry for containers (vs the older google container registry)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>these four cells should be collapsed into one</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>Vertex AI SDK</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>Vertex AI TensorBoard</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <h4>Option: Use a Previously Created Vertex Tensorboard Instance  -&gt; Option: Use a previously created Vertex AI TensorBoard instance</h4><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <p>Vertex AI SDK</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

Comment (andrewferlitsch) [inline notebooks/official/pytorch_image_classification_distributed_data_parallel_training_with_vertex_sdk/multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb:1]: <h3>Training Output Artifact -&gt; View output artifact from training</h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/489/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

**Comment by andrewferlitsch** [comment]
This is failing tests because it is testing other unrelated notebooks:

 get_started_vertex_experiments.ipynb                                  PASSED    00:03:46    https://console.cloud.google.com/cloud-build/builds/ff669a3e-9456-4f56-907a-557066a53aae?project=1012616486416
Step #4: sdk-metric-parameter-tracking-for-custom-jobs.ipynb                   PASSED    00:17:08    https://console.cloud.google.com/cloud-build/builds/c8bc125c-4f18-4c85-bbe2-59531005d6cf?project=1012616486416
Step #4: sdk-metric-parameter-tracking-for-locally-trained-models.ipynb        PASSED    00:04:51    https://console.cloud.google.com/cloud-build/builds/77557d10-7593-4f1c-a56c-cf65061f9710?project=1012616486416
Step #4: google_cloud_pipeline_components_model_upload_predict_evaluate.ipynb  PASSED    00:34:25    https://console.cloud.google.com/cloud-build/builds/17e904f3-3499-44c6-9447-591c81bedbb8?project=1012616486416
Step #4: pipelines_intro_kfp.ipynb                                             PASSED    00:06:25    https://console.cloud.google.com/cloud-build/builds/969521a9-df47-4092-bf82-cbc2c8cf519e?project=1012616486416
Step #4: get_started_bq_datasets.ipynb                                         FAILED    00:07:39    https://console.cloud.google.com/cloud-build/builds/4b60153e-c2e9-4265-94c0-3071bc6d5336?project=1012616486416
Step #4: get_started_vertex_datasets.ipynb                                     FAILED    00:03:32    https://console.cloud.google.com/cloud-build/builds/ffec1cad-f537-4a61-b32f-09b175938c22?project=1012616486416
Step #4: get_started_automl_training.ipynb                                     FAILED    02:29:05    https://console.cloud.google.com/cloud-build/builds/8f4be373-1281-4889-be88-bbb82b16699e?project=1012616486416
Step #4: get_started_bqml_training.ipynb                                       FAILED    00:12:54    https://console.cloud.google.com/cloud-build/builds/6b40b25e-a626-4f23-abcb-35def46332ad?project=1012616486416
Step #4: get_started_vertex_training.ipynb                                     FAILED    00:22:03    https://console.cloud.google.com/cloud-build/builds/0fa47cd5-da01-4e3e-8bbe-646af673c934?project=1012616486416
Step #4: get_started_with_cmek_training.ipynb                                  FAILED    00:02:45    https://console.cloud.google.com/cloud-build/builds/fa5592fd-a86a-4a08-a509-0e1be8acdb3a?project=1012616486416
Step #4: mlops_experimentation.ipynb                                           FAILED    00:02:54    https://console.cloud.google.com/cloud-build/builds/7c62b99c-c785-4e1a-a296-778460f9379c?project=1012616486416
Step #4: get_started_with_dataflow_pipeline_components.ipynb                   FAILED    00:02:48    https://console.cloud.google.com/cloud-build/builds/69ea9e1b-d3f7-492c-919d-b922c5803613?project=1012616486416
Step #4: get_started_with_vertex_private_endpoints.ipynb                       FAILED    00:06:13    https://console.cloud.google.com/cloud-build/builds/c34dede5-4839-4560-81cf-782e708044a4?project=1012616486416
Step #4: get_started_with_tf_serving.ipynb                                     FAILED    00:03:41    https://console.cloud.google.com/cloud-build/builds/a9b6db7b-3e4a-4fc0-90f1-0bd224c4bf8e?project=1012616486416
Step #4: sdk_automl_tabular_forecasting_batch.ipynb                            FAILED    01:33:21    https://console.cloud.google.com/cloud-build/builds/f660b309-63f4-4f81-95f5-65f5954858a3?project=1012616486416
Step #4: multi_node_ddp_gloo_vertex_training_with_custom_container.ipynb       FAILED    00:03:00    https://console.cloud.google.com/cloud-build/builds/82408d57-d2a9-48da-aef6-ba3da48470db?project=1012616486416
Step #4:

**Comment by kweinmeister** [comment]
Build error: `File `'trainer/task.py'` not found.`

**Comment by andrewferlitsch** [comment]
@ivanmkc tests not ran: Changed files did not match file filter

**Review by andrewferlitsch — APPROVED** [review]

**Comment by andrewferlitsch** [comment]
was passing tests before I fixed a couple of non-code titles. Now says internal error. Will ignore and merge.

**Pull request merged**